### PR TITLE
Lab1 Polina Minina

### PR DESCRIPTION
**Pull request by pminina01**

### CODE REVIEW

**Review by sevenzing — APPROVED** [review]

**Review by bragov4ik — APPROVED** [review]

File: app_go/app.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        loc, _ := time.LoadLocation("Europe/Moscow")
+        now := time.Now().In(loc).String()
+        w.Write([]byte(fmt.Sprintf("<h1>Moscow time is %s</h1>", now)))
+}
+
+func main() {
+        port := "3000"
+        mux := http.NewServeMux()
+        mux.HandleFunc("/", indexHandler)
+        http.ListenAndServe(":"+port, mux)
```
Comment (Ilya-Kolomin) [inline app_go/app.go:19]: Better practice to add error handling and logging it.

File: app_python/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ```bash
+    python3 -m venv <myenvname>  
+    source <myenvname>/bin/activate  
+    pip install -r requirements.txt  
+    ```
+
+4. Run the following command to start the application
+
+    ```bash
+    flask run --host=127.0.0.1 --port=5000
```
Comment (Ilya-Kolomin) [inline app_python/README.md:55]: Better to not use flask development server for production.

**Review by Ilya-Kolomin — APPROVED** [review]
Review OK👍

**Review by Feyd-Rauth — APPROVED** [review]

File: app_go/app.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+        "fmt"
+        "net/http"
+        "time"
+)
+
+func indexHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (Anatolay) [inline app_go/app.go:9]: Avoid single-letter variable names

File: app_python/PYTHON.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* Use virtual environment
+
+## Framework choice - Flask
+
+### Pros
+* Simple for our goal
+* Flexible
+
+### Cons
+* Too simple
```
Comment (Anatolay) [inline app_python/PYTHON.md:17]: What is the disadvantage exactly? Lack of features, lack of extensibility, lack of community support?

File: app_python/app.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+TIMEZONE = "Europe/Moscow"
+TIME_FORMAT = "%d.%m.%Y %H:%M:%S"
+
+app = Flask(__name__)
+
+
+@app.route("/")
+def show_time():
+    moscow_time = datetime.now(pytz.timezone(TIMEZONE)).strftime(TIME_FORMAT)
+    html = "<center>" + moscow_time + "</center>"
```
Comment (Anatolay) [inline app_python/app.py:14]: Avoid hardcoding html, better use a template engine

**Review by Anatolay — CHANGES_REQUESTED** [review]

**Pull request closed without merging**